packages/fastify/src/index.ts

```diff
@@ -1,14 +1,25 @@
-import { createClerkClient } from '@clerk/clerk-sdk-node';
-import type { FastifyPluginCallback } from 'fastify';
-
-const clerkClient = createClerkClient({});
-
-export const plugin: FastifyPluginCallback = (instance, opts, done) => {
-  instance.decorate('auth', {});
-  instance.get('/', (request, reply) => {
-    console.log(clerkClient);
-    console.log(request.url);
-    reply.serialize({ hello: 'there' });
-  });
+import type { WithAuthProp } from '@clerk/clerk-sdk-node';
+import type { FastifyInstance, FastifyPluginCallback, FastifyRequest } from 'fastify';
+import fp from 'fastify-plugin';
+
+import { polyfillServerResponseMethods } from './polyfill';
+import { withClerkMiddleware } from './withClerkMiddleware';
+
+const plugin: FastifyPluginCallback = (instance: FastifyInstance, opts, done) => {
+  polyfillServerResponseMethods(instance);
+  instance.decorateRequest('auth', null);
+  // run clerk as a middleware to all scoped routes
+  instance.addHook('preHandler', withClerkMiddleware(opts));
+
   done();
 };
+
+export const clerkPlugin = fp(plugin, {
+  name: '@clerk/fastify',
+  fastify: '4.x',
+});
+
+export const getAuth = (req: FastifyRequest) => {
+  const authReq = req as WithAuthProp<FastifyRequest>;
```

Comment:
Shall we add an error if someone tries to use getAuth without registering clerkPlugin or if they try to use it in a route not covered by clerkPlugin?

Reply:
do you have in mind an example of the thrown error message?